systems/plants/TimeSteppingRigidBodyManipulator.m

```diff
@@ -605,11 +605,9 @@ function checkDirty(obj)
               obj.LCP_cache.data.fastqp_active_set = [];
 
               % take M_active, z_inactive on reduced LCP
-              obj.LCP_cache.data.M_active = false(nL+nP+(2+mC)*nC,1);
-              obj.LCP_cache.data.M_active(z_inactive_guess) = ...
-                M(z_inactive_guess,z_inactive_guess)*z(z_inactive_guess)+w(z_inactive_guess)<1e-8;
               obj.LCP_cache.data.z_inactive = false(nL+nP+(2+mC)*nC,1);
               obj.LCP_cache.data.z_inactive(z_inactive_guess) = z(z_inactive_guess)>lb(z_inactive_guess)+1e-8;            
+              obj.LCP_cache.data.M_active = obj.LCP_cache.data.z_inactive;
             end
 %             if isempty(active_set_fail_count)
```

Comment:
ok.  maybe add some of this as a comment?  (because it looks like a bug :).

On Jan 6, 2015, at 11:31 AM, Scott Kuindersma <notifications@github.com> wrote:

> In systems/plants/TimeSteppingRigidBodyManipulator.m:
> 
> >                obj.LCP_cache.data.z_inactive = false(nL+nP+(2+mC)*nC,1);
> >                obj.LCP_cache.data.z_inactive(z_inactive_guess) = z(z_inactive_guess)>lb(z_inactive_guess)+1e-8;            
> > +              obj.LCP_cache.data.M_active = obj.LCP_cache.data.z_inactive;
> >              end
> >  %             if isempty(active_set_fail_count)
> No because you still have problem #2. In practice sometimes it's 1e-7, 1e-6, etc. Doing things the tolerance-free way amounts to at most a couple extra iterations inside fastQP, but a guarantee that you wont be solving the wrong problem.
> 
> —
> Reply to this email directly or view it on GitHub.
>